### ISSUE TITLE
Incorrect parsing of os-release file

### ISSUE THREAD

**Issue by natanjunges** [PJ-Singh-001/Cubic#273]
**Describe the bug**

I have an os-release file with a comment line, which is perfectly legal according to `man os-release`, but Cubic fails reading it. By reading the code on how it parses the file, I think it would also fail with an empty line, which is, again, perfectly legal.

**To Reproduce**

Comment a line in /etc/os-release and try to generate the customized disk image.

**Expected behavior**

It should be able to parse the file properly, following what `man os-release` specifies.

**OS Information (please complete the following information):**
 - OS/Distro Name: Ubuntu
 - OS Version: 22.04

**Cubic Information (please complete the following information):**
 - Cubic Version: 2023.09.84
 - ISO Customizing: ubuntu-22.04.3-desktop-amd64.iso
 - Download Link:  https://releases.ubuntu.com/22.04/ubuntu-22.04.3-desktop-amd64.iso.torrent
 
**Cubic Log:**

```
Exception in thread Thread-9 (navigate):
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/share/cubic/cubic/navigator.py", line 613, in navigate
    result = new_page.enter(action, page) if new_page else None
  File "/usr/share/cubic/cubic/pages/generate_page.py", line 144, in enter
    is_error = copy_kernel_files()
  File "/usr/share/cubic/cubic/pages/generate_page.py", line 377, in copy_kernel_files
    is_pop_os_based = constructor.os_is_distribution('pop', model.project.custom_root_directory)
  File "/usr/share/cubic/cubic/utilities/constructor.py", line 246, in os_is_distribution
    key, value = line.split('=', 1)
ValueError: not enough values to unpack (expected 2, got 1)
```

**Possible solution**

This might be related to #221 and #247. My solution is to follow the examples on how to parse the os-release file using python provided by `man os-release`.

**Comment by PJ-Singh-001** [comment]
Would you please share your `/etc/os-release` file with the comment?

**Comment by natanjunges** [comment]
```
PRETTY_NAME="bleUX 0.11.0 (Jammy Jellyfish)"
NAME="bleUX"
# VERSION_CODENAME in a format suitable for presentation to the user
CODENAME="Jammy Jellyfish"
VERSION_ID=0.11.0
VERSION="0.11.0 (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=bleux
ID_LIKE="ubuntu debian"
HOME_URL="https://github.com/bleux-os/ubuntu"
DOCUMENTATION_URL="https://github.com/bleux-os/ubuntu/wiki"
BUG_REPORT_URL="https://github.com/bleux-os/ubuntu/issues"
UBUNTU_CODENAME=jammy
```

**Comment by PJ-Singh-001** [comment]
Thanks. I'll take a look at this.

**Comment by PJ-Singh-001** [comment]
@natanjunges,

Thank you for a well reported bug, with all requested information, and for pointing to relevant documentation!

I'll try to get a fix out soon.

**Comment by PJ-Singh-001** [comment]
Please let me know if the following patch woks?

 1. Copy the following text and save it to a file named `constructor.patch`.

        *** constructor.original.py	2023-10-31 00:14:56.769665241 -0400
        --- constructor.py	2023-12-06 21:06:26.112335337 -0500
        ***************
        *** 211,215 ****
                          key, value = line.split('=', 1)
        !                 key = key.upper() if key else None
                          if key == 'ID':
        !                     distribution = value.rstrip().lower() if value else None
                              break
        --- 211,215 ----
                          key, value = line.split('=', 1)
        !                 key = key.strip().upper() if key else None
                          if key == 'ID':
        !                     distribution = value.strip().lower() if value else None
                              break
        ***************
        *** 245,249 ****
                      for line in file:
                          key, value = line.split('=', 1)
        !                 key = key.upper() if key else None
        !                 value = value.rstrip().lower() if value else None
                          if key == 'ID':
        --- 245,252 ----
                      for line in file:
        +                 line = line.strip()
        +                 if not line: continue
        +                 if line.startswith('#'): continue
                          key, value = line.split('=', 1)
        !                 key = key.strip().upper() if key else None
        !                 value = value.strip().lower() if value else None
                          if key == 'ID':

 2. Execute the following command to apply the patch:

        cd <directory where you saved constructor.patch>
        sudo patch /usr/share/cubic/cubic/utilities/constructor.py constructor.patch

**Comment by natanjunges** [comment]
It works.

**Comment by PJ-Singh-001** [comment]
Thank you for testing.
I will issue a new release this month with the fix.

**Comment by PJ-Singh-001** [comment]
Fix released in Cubic version 2023.12-85.
